azurerm/resource_arm_virtual_machine.go

```diff
@@ -1557,3 +1496,53 @@ func findStorageAccountResourceGroup(meta interface{}, storageAccountName string
 
 	return id.ResourceGroup, nil
 }
+
+func resourceArmVirtualMachineStorageOsProfileHash(v interface{}) int {
```

Comment:
Just a note here: the implicit `schema.HashResource` has been working pretty well for me for most complex sets I've been doing. Unless there's specific rules that this structure needs to be abide by for hashing to remove some factors from the algorithm, you might be better off just using that. Can help get rid of some more code too :slightly_smiling_face:

Reply:
Yeah I was looking into that, unfortunately there's some fields missing from some of the Hash functions (hence why I had to change some from Set -> List -> Set) - and then just relocated them in the file :(